### PR TITLE
Update README.me with documentation for #3

### PR DESCRIPTION
**Pull request by inlined**
Front-page usage stuff. Not sure if this is better in the main README or if I should put this in cloud/. I'm leaning towards putting it here because auth/push won't have their own directories.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,21 +15,135 @@ ready until after the talk.
...
+individual ParseObject would map to the Firebase Realtime
+Database. We offer two files to help you migrate from Parse:
+
+ 1. `migrator.js` A tool that allows you two keep your existing Cloud Code *and* use Cloud Code to migrate data out of Parse.
+ 2. `firebase-rest.js` A simple wrapper that helps you talk to Firebase in Cloud Code (where the Firebase Node module doesn't work).
+ 
+Imagine you have a `BlogPost` class in Parse with the following fields:
+
+* **title** *(String)* Name the post
+* **body** *(String)* Blog contets
```
Comment (bklimt) [inline README.md]: contents

File: README.md [hunk 2]
```diff
@@ -15,21 +15,135 @@ ready until after the talk.
...
+* **keywords** *(String)* Originally you chose to do text searches on a space separated list
+* **kewords_list** *(Array)* You realized later that Parse can search an array better and added Cloud Code to put **keywords** into **keywords_list**
```
Comment (bklimt) [inline README.md]: s/kewords/keywords/

File: README.md [hunk 3]
```diff
@@ -15,21 +15,135 @@ ready until after the talk.
...
+
+A logical structure for your blog posts in Firebase might look like:
+
+```json
+{
+  "posts": {
+    "aBlogPostID": {
+      "title": "Firebase expands to become a unified app platform",
+      "body": "Eighteen months ago, Firebase joined Google...",
+      "keywords": ["googleio", "announcements"],
```
Comment (bklimt) [inline README.md]: Why is this here? Also, you said above that "keywords" was a String...

File: README.md [hunk 4]
```diff
@@ -15,21 +15,135 @@ ready until after the talk.
...
+    "announcements": {
+      "aBlogPostID": true
+    }
+  }
+}
+```
+
+A few things are noticeably different in this structure:
+
+1. We no longer use the legacy `"keywords"` array
```
Comment (bklimt) [inline README.md]: But it's still there? And I thought it was a String before...

File: README.md [hunk 5]
```diff
@@ -15,21 +15,135 @@ ready until after the talk.
...
+  url: 'https://my-firebase-database-url.firebaseio.com',
+  key: 'if-my-firebase-database-is-secured-this-key-gives-me-admin-access',
+  parse: Parse
+});
+
+// Notice this is almost the same (and actually a bit simpler) than the original beforeSave
+migrator.beforeSave('BlogPost', function(request) {
+  var object = request.object;
+  if (object.isDirty('keywords_list')) {
+    throw 'keywords_list is managed by Cloud Code only';
```
Comment (bklimt) [inline README.md]: Are you sure this works as intended? I seem to recall jQuery promises (and therefore Parse.Promise) being weird about "throw".

File: README.md [hunk 6]
```diff
@@ -15,21 +15,135 @@ ready until after the talk.
...
+  
+  return Parse.Promise.when(saveObject, ref.patch(keywordsUpdate));
+});
+
+// This is run between beforeDelete and afterDelete.
+migrator.migrateDelete('BlogPost', function(object) {
+  // Setting a value to null in Firebase deletes it
+  var updates = {};
+  _.each(object.get('keywords'), function(keyword) {
+    updates['keywords/' + keyword + '/' + object.id] = null;
```
Comment (bklimt) [inline README.md]: Ah, I would note above when you put "keywords" in your firebase schema that it exists solely to make deleting objects more efficient.

**Comment by bklimt** [comment]
LGTM, other than the nits above.

File: README.md (hunk 4 above)
Comment (inlined) [inline README.md]: Whoops. We no longer use the legacy keywords string.

File: README.md (hunk 5 above)
Comment (inlined) [inline README.md]: Maybe it was fixed? Unit tests register functions that throw (see 'beforeSave' > 'should not run migrations for failed saves')
